src/Native/gen-buildsys-win.bat

```diff
@@ -28,7 +28,9 @@ popd
 :DoGen
 if /i "%3" == "x64"     (set __ExtraCmakeParams=%__ExtraCmakeParams% -A x64)
 if /i "%3" == "x86"     (set __ExtraCmakeParams=%__ExtraCmakeParams% -A Win32)
-"%CMakePath%" "-DCMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE%" "-DCMAKE_INSTALL_PREFIX=%__CMakeBinDir%" "-DMKL_LIB_PATH=%MKL_LIB_PATH%" -G "Visual Studio %__VSString%" %__ExtraCmakeParams% -B. -H%1
+if /i "%3" == "arm64"     (set __ExtraCmakeParams=%__ExtraCmakeParams% -A arm64)
+if /i "%3" == "arm"     (set __ExtraCmakeParams=%__ExtraCmakeParams% -A arm)
+"%CMakePath%" "-DCMAKE_BUILD_TYPE=%CMAKE_BUILD_TYPE%" "-DCMAKE_INSTALL_PREFIX=%__CMakeBinDir%" "-DMKL_LIB_PATH=%MKL_LIB_PATH%" "-DARCHITECTURE=%3" -G "Visual Studio %__VSString%" %__ExtraCmakeParams% -B. -H%1
```

Comment:
Why do we need both `-DARCHITECTURE` and `-A` above? Can we just use 1 command line arg to set the archiecture?

Reply:
So the `-A` doesn't work on the Unix Makefile Generator, but that is how you set the architecture correctly for visual studio. I needed a way in the cmakelists file to be able to exclude native projects for both generators, and thats how I ended up with the `-DARCHITECTURE`. I'm sure there is a way I could use only the `-A` in visual studio and not need the `-DARCHITECTURE`, I just haven't been able to figure it out yet. We will always need `-DARCHITECTURE` for the Unix generator, its just if we can remove it from the visual studio one.

Reply:
Have you looked how it is done in other repos? For example `dotnet/runtime`?

Reply:
I'm checking that out right now. They pass the `-arch` flag. Thats a custom flag. I'm still investigating how they plumb it internally.

Reply:
That eventually turns into this `/p:TargetArchitecture=$arch`. Still looking into how thats passed to the native side of things.

Reply:
So it actually looks like they do the same thing. 
https://github.com/dotnet/runtime/blob/e4b4807e2fae2164d9116fbcdd49ba9044461e7e/eng/native/gen-buildsys.cmd#L34 they set the `-A` same way I do.

https://github.com/dotnet/runtime/blob/e4b4807e2fae2164d9116fbcdd49ba9044461e7e/src/coreclr/build-runtime.cmd#L450 they a flag `-DCLR_CMAKE_TARGET_ARCH` which is used the same why I am using the `ARCHITECTURE` flag. They do more complex stuff with it than we do since they target so many things, but its the same idea.

Reply:
Sounds good. Thanks for verifying.